content/blog/browser-nativefs/index.md

```diff
@@ -0,0 +1,365 @@
+[`URL.createObjectURL()`](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL)
+method.
+
+<blockquote>
+  To prevent memory leaks, always revoke the URL after the download.
+</blockquote>
+
+```js
+const saveFile = async (blob) => {
+  const a = document.createElement("a");
+  a.download = "my-file.txt";
+  a.href = URL.createObjectURL(blob);
+  a.addEventListener("click", (e) => {
+    setTimeout(() => URL.revokeObjectURL(a.href), 30 * 1000);
+  });
+  a.click();
+};
+```
+
+### The problem
+
+A massive downside of the _download_ approach is that there is no way to make a classic
+open→edit→save flow happen, that is, there is no way to _overwrite_ the original file.
+Instead, you end up with a new _copy_ of the original file
+in the operating system's default Downloads folder whenever you "save".
+
+<h2 id="the-file-system-access-api">The File System Access API</h2>
+
+The File System Access API makes both operations, opening and saving, a lot simpler.
+It also enables _true saving_, that is, you can not only choose where to save a file,
+but also overwrite an existing file.
+
+<blockquote>
+  For a more thorough introduction to the File System Access API, see the article
+  <a href="https://web.dev/file-system-access/">The File System Access API: simplifying access to local files</a>.
+</blockquote>
+
+### Opening files
+
+With the [File System Access API](https://wicg.github.io/file-system-access/),
+opening a file is a matter of one call to the `window.showOpenFilePicker()` method.
+This call returns a file handle, from which you can get the actual `File` via the `getFile()` method.
+
+```js
+const openFile = async () => {
+  try {
+    // Always returns an array.
+    const [handle] = await window.showOpenFilePicker();
+    return handle.getFile();
+  } catch (err) {
+    console.error(err.name, err.message);
+  }
+};
+```
+
+### Opening directories
+
+Open a directory by calling
+`window.showDirectoryPicker()` that makes directories selectable in the file dialog box.
+
+### Saving files
+
+Saving files is similarly straightforward.
+From a file handle, you create a writable stream via `createWritable()`,
+then you write the Blob data by calling the stream's `write()` method,
+and finally you close the stream by calling its `close()` method.
+
+```js
+const saveFile = async (blob) => {
+  try {
+    const handle = await window.showSaveFilePicker({
+      types: [
+        {
+          accept: {
+            // Omitted
+          },
+        },
+      ],
+    });
+    const writable = await handle.createWritable();
+    await writable.write(blob);
+    await writable.close();
+    return handle;
+  } catch (err) {
+    console.error(err.name, err.message);
+  }
+};
+```
+
+## Introducing browser-nativefs
+
+As perfectly fine as the File System Access API is,
+it's [not yet widely available](https://caniuse.com/#feat=file-system-access-api).
```

Comment:
https://caniuse.com/native-filesystem-api

this seems broken: https://caniuse.com/#feat=file-system-access-api

Reply:
Oh, they accepted my PR to change the name, but left the link target intact. Fair. Thanks!